worlds/psychonauts/Rules.py

```diff
@@ -0,0 +1,506 @@
+        return state.has(ItemName.Flowers, self.player)
+
+    def has_plunger(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Plunger, self.player)
+
+    def has_hedge_trimmers(self, state: CollectionState) -> bool:
+        return state.has(ItemName.HedgeTrimmers, self.player)
+
+    def has_rolling_pin(self, state: CollectionState) -> bool:
+        return state.has(ItemName.RollingPin, self.player)
+
+    def has_candle(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Candle, self.player)
+
+    def has_both_candles(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Candle, self.player, 2)
+
+    def has_megaphone(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Megaphone, self.player)
+
+    def has_freds_letter(self, state: CollectionState) -> bool:
+        return state.has(ItemName.FredsLetter, self.player)
+
+    def has_priceless_coin(self, state: CollectionState) -> bool:
+        return state.has(ItemName.PricelessCoin, self.player)
+
+    def has_musket(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Musket, self.player)
+
+    def has_cobweb_duster(self, state: CollectionState) -> bool:
+        return state.has(ItemName.CobwebDuster, self.player)
+
+    def has_levitation(self, state: CollectionState) -> bool:
+        if self.world.options.StartingLevitation:
+            return True
+        else:
+            return state.has(ItemName.Levitation, self.player)
+
+    def has_telekinesis(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Telekinesis, self.player)
+
+    def has_pyrokinesis(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Pyrokinesis, self.player)
+
+    def has_clairvoyance(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Clairvoyance, self.player)
+
+    def has_marksmanship(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Marksmanship, self.player)
+
+    def has_invisibility(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Invisibility, self.player)
+
+    def has_shield(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Shield, self.player)
+
+    def has_confusion(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Confusion, self.player)
+
+    def has_upper_asylum_access(self, state: CollectionState) -> bool:
+        return state.has_all([ItemName.LobotoPainting, ItemName.GloriasTrophy, ItemName.StraightJacket], self.player)
+
+    def has_oleander_boss_access(self, state: CollectionState) -> bool:
+        return state.has_all([ItemName.Cake, ItemName.Telekinesis, ItemName.Pyrokinesis], self.player)
+
+    def redeemed_brain_goal(self, state: CollectionState, amount) -> bool:
+        return amount <= sum([state.has(item_name, self.player) for item_name in BRAIN_JARS])
```

Comment:
This wasn't done at the time because `has_from_list_unique` was added in AP 0.5.0 which was only a release candidate at the time, so making the change would have broken compatibility with AP 0.4.6. This change can be made now.